userspace/otpilot/src/spi_processor.rs

```diff
@@ -269,6 +283,11 @@ impl<'a> SpiProcessor<'a> {
                 }
                 self.clear_device_status(true, true)
             }
+            OpCode::ReadExtendedAddressRegister => {
+                // We can't handle this, but we also don't want it to raise an
+                // UnsupportOpCode error.
+                self.ignore_transaction()
```

Comment:
Would it make sense to spew something here, so that an unsuspecting victim will see evidence that the command they sent is getting ignored?

Reply:
You could make the argument that we generally want to print information about unsupported commands. I found this particular one by testing against u-boot running on Nuvoton. I'm not why it's making this call since the extended address register is not supported in Haven. Looks like a bug in the Nuvoton u-boot code to me, really. Since it's not supported, it'll return `0xff` on SPI. I believe that works exactly the same as if this was attempted with a regular SPI flash chip that doesn't support the command.

Reply:
With that, I think you're right, we should not just silently ignore it. I removed the special handling code for the `ReadExtendedAddressRegister` again. This results in `process_spi_header()` returning an error which is then handled (by printing) in the `run()` loop. I think this is actually cleaner.